src/utils.sh

```diff
@@ -8,51 +8,96 @@ VERSION_TAG="v0.0.0-rc0"
 # create flag:variable_name dictionary
 declare -A flag_to_variable_dict
 
-add_option_args() {
+######################################
+# add_option_args
+#
+#    declare a valid command-line argument(s)
+#
+# ARGUMENTS:
+#    OPTION_NAME    The name of the argument (e.g "VERBOSE")
+#    a list of one or more argument switches (e.g "-v", "--verbose")
+# OUTPUTS:
+#    Write output to stdout
+# RETURN:
+#
+######################################
+
+function add_option_args() {
     if [[ $# > 1 ]];
     then
-        local option_flag=$1; shift
-        local var_name=$1; shift
-        flag_to_variable_dict[$option_flag]=$var_name
+        local option_name=$1; shift
+        while [ $# -ne 0 ];
+        do
+            flag_to_variable_dict[$1]=$option_name
+            shift
+        done
     fi
 }
 
-# generic command line parser
+######################################
+# clear_option_args
+#
+#    clears the list of valid command-line arguments
+#
+# ARGUMENTS:
+#
+# OUTPUTS:
+#
+# RETURN:
+#
+######################################
+
+function clear_option_args() {
+    flag_to_variable_dict=()
+}
+
 function cmd_parser() {
-    # create flag:variable_name dictionary
+    # create flag:variable_name dictionary and initialize to empty string
     declare -A parsed_cmd
+    for key in ${!flag_to_variable_dict[*]};
+    do
+        parsed_cmd[${flag_to_variable_dict[$key]}]=""
+    done
 
-    for arg in "$@"
+    while [ $# -ne 0 ];
```

Comment:
for my own learning, is it better to use a while instead of a for loop in these kind of situations?

Reply:
1- Since I need to look at $1 and $2, the $arg and $2 seemed inconsistent.
2- I personally find this form simpler and less error prone.